Build error with clang-15 and clang-14

when using this library with clang-14 or clang-15 the following error occurs:
```
In file included from /builds/build/deps/asiochan/include/asiochan/asiochan.hpp:5:
In file included from /builds/build/deps/asiochan/include/asiochan/channel.hpp:9:
In file included from /builds/build/deps/asiochan/include/asiochan/detail/channel_method_ops.hpp:11:
In file included from /builds/build/deps/asiochan/include/asiochan/select.hpp:17:
In file included from /builds/build/deps/asiochan/include/asiochan/detail/select_impl.hpp:14:
/builds/build/deps/asiochan/include/asiochan/select_result.hpp:100:18: error: atomic constraint must be of type 'bool' (found 'const auto')
        requires is_alternative<no_result_t>
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/include/asiochan/select_result.hpp
+++ b/include/asiochan/select_result.hpp
@@ -32,7 +32,7 @@ namespace asiochan
         using discriminator_type = std::size_t;
 
         template <typename T>
-        static constexpr auto is_alternative = (std::same_as<T, typename Ops::result_type> or ...);
+        static constexpr bool is_alternative = (std::same_as<T, typename Ops::result_type> or ...);
 
         template <std::convertible_to<variant_type> T>
         select_result(T&& value, discriminator_type const alternative)

